### PR TITLE
Make RpcTest class re-usable by other RPC backends

### PR DESCRIPTION
**Pull request by xush6528**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#27199 Make RpcTest class re-usable by other RPC backends**

# Problem

Want to pass args to other RPC backends through init_method.

# Solution

Set up init_method in rpc tests.

Differential Revision: [D17069062](https://our.internmc.facebook.com/intern/diff/D17069062/)

### CODE REVIEW

**Comment by xush6528** [comment]
Abandon #27114, because I can't export to that PR.

**Comment by pietern** [comment]
@xush6528 What happened?

File: test/rpc_test.py [hunk 1]
```diff
@@ -168,42 +174,38 @@ def test_duplicate_name(self):
                 self_name="duplicate_name",
                 backend=BACKEND,
                 self_rank=self.rank,
-                init_method=RPC_INIT_URL,
+                init_method=init_method,
```
Comment (pietern) [inline test/rpc_test.py]: Remove kwarg if it's an empty string?

File: test/rpc_test.py [hunk 2]
```diff
@@ -168,42 +174,38 @@ def test_duplicate_name(self):
...
-        dist.init_process_group(
-            backend="gloo", rank=self.rank, world_size=self.world_size, store=store
-        )
-        dist.init_model_parallel(
-            self_name="worker{}".format(self.rank),
-            backend=BACKEND,
-            self_rank=self.rank,
-            init_method=RPC_INIT_URL,
-        )
+        # dist.init_model_parallel(...) called in `@_wrap_with_rpc` first.
```
Comment (pietern) [inline test/rpc_test.py]: Please revert. This test depends on the calls in both places to be the same.

File: test/rpc_test.py [hunk 3]
```diff
@@ -168,42 +174,38 @@ def test_duplicate_name(self):
...
     def test_init_invalid_backend(self):
+        init_method = ""
+
         with self.assertRaisesRegex(RuntimeError, "Unrecognized RPC backend"):
             dist.init_model_parallel(
                 self_name="worker{}".format(self.rank),
                 backend="invalid",
                 self_rank=self.rank,
-                init_method=RPC_INIT_URL,
+                init_method=init_method,
```
Comment (pietern) [inline test/rpc_test.py]: Same, kwarg can be removed.

File: test/rpc_test.py [hunk 4]
```diff
@@ -224,7 +226,7 @@ def test_invalid_names(self):
                 self_name="".join(["a" for _ in range(500)]),
                 backend=BACKEND,
                 self_rank=self.rank,
-                init_method=RPC_INIT_URL,
+                init_method=init_method,
```
Comment (pietern) [inline test/rpc_test.py]: Idem.

**Comment by xush6528** [comment]
@pietern 

Abandon this one. I just found ghexport exported a wrong version of code.

**Comment by xush6528** [comment]
Has problem exporting code. Abandon this one. Move to #27208.

**Pull request closed without merging**